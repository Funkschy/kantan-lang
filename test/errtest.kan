import "other";
import "io";
import "files/empty";

io.1;
io.test.1;
io.printf;

type Other struct {
    o: Other2
}
type Other2 struct {
    o: Other
}

type S1 struct {
    f: other.MyStruct,
    i: i32
}

type Test struct {
    test_ident: TestType,
    other: *Test
}

type TestType struct {}

let test = Test {
    test_ident: TestType {},
    other: null
};

Test.other;
test.test_ident;
test.kekw;

test as *TestType;
let test: i32 = 0;

return 1;
delete 1;

1 == !1 && 1 == 1 || (-1 + -1) != 0;

1 >= 0;
&1 + *(1 + 1 % 2);

if true {
    let s = "";
} else if false {

} else {

}

while true {
    let test = "";
}

0 - (2;; 420 - (69 / 1); (); 42 520;
let i = 52;
i = other = 2;


type One struct {
    two: Two
}

type Two struct {
    three: *Three
}

type Three struct {
    i: i32
}

def function(one: *One, t: Test): other.MyStruct {
    let test = 1;
    1 + 1;
    one.two.three.i.error;
    t.other;
}

let other = 0;
other;
let あ　= 0;
import "files/empty";

'';
't';
'te';

