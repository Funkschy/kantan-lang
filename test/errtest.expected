# ret 6
Compiling ./errtest.kan
6 errors found
ERROR: Expected ')', but got ';'
--> ./errtest.kan:36:7

ERROR: Could not parse statement: 
();
--> ./errtest.kan:36:26

ERROR: Expected ';', but got '520'
--> ./errtest.kan:36:33

ERROR: Unknown symbol: 'あ'
--> ./errtest.kan:48:5

ERROR: Char literal with length 0: ''
--> ./errtest.kan:51:2

ERROR: Char literal with length 2: 'te'
--> ./errtest.kan:53:2

import 'test'
import 'oof'
1 functions
def function: *test.Test
is expr: 1
Test{test_ident: TestType{field: span.get((&span_interner), function((420-(69/1)), test.i.other)).len}, other: null}

